iodata/formats/xyz.py

```diff
@@ -91,15 +91,129 @@
 """
 
 
+BOOL_MAP = {"T": True, "F": False, "True": True, "False": False}
+
+
+def _convert_title_value(value: str):
+    """Search for the correct dtype and convert the string."""
+    list_of_splits = value.split()
+    # If it is just one item, first try int, then float and finally return a bool or str
+    if len(list_of_splits) == 1:
+        value = value.strip()
+        if value.isdigit():
+            converted_value = int(value)
+        else:
+            try:
+                converted_value = float(value)
+            except ValueError:
+                if value in BOOL_MAP.keys():
+                    converted_value = BOOL_MAP[value]
+                else:
+                    converted_value = value
```

Comment:
It seems one of my comments was nixed by github somehow. It is minor, just for readability. The chain of ifs and try-excepts can be flattened as follows:

```suggestion
        # int
        if value.isdigit():
            return int(value)
        # float
        try:
            return float(value)
        except ValueError:
            pass
        # bool
        if value in BOOL_MAP.keys():
            return BOOL_MAP[value]
        # str
        return value
```

(idem for second half)

Reply:
This looks nicer indeed. However, I see two (minor) problems arise. 

1. Roberto complains about having a function with more than 6 return statements, which can be solved by adding two extra helper function: `_convert_single_value` and `_convert_multiple_values`. 

2. I don't think a simple `if` statement is possible to check whether all elements in an array are digits. The easiest thing I can come up with is something like this: 

```python
if np.all([True if b.isdigit() else False for b in list_of_splits.split()]):
    return np.array(list_of_splits, dtype=np.int)
``` 

but I'm not sure if this is a cleaner solution.

Reply:
Too many returns may be confusing too. Just leave it like it is. The suggestion is not worth the trouble.